lib/impl/AzureBlobFiles.js

```diff
@@ -205,10 +205,17 @@ class AzureBlobFiles extends Files {
     }
 
     if (credentials || !this.hasOwnCredentials) {
+      // this seems like an error, if we get here without credentials this will throw
```

Comment:
On line 202 the if statement re-generates new credentials so it should be fine. So the only case where we could get into that condition with `credentials=false` would be if `this.hasOwnCredentials=false` but that would have thrown on Init. But yes I think you are right and `&&` would be more correct.

@sandeep-paliwal wdyt?